From dcf520da15d940c900d7e8ffd5a9b05427c54dc8 Mon Sep 17 00:00:00 2001
From: Philip Chimento <philip.chimento@gmail.com>
Date: Wed, 5 Jul 2017 22:47:44 -0700
Subject: [PATCH 4/9] headers: Fix symbols visibility

Some symbols that need to be public are not marked as such.
---
 js/public/Utility.h | 2 +-
 js/src/jsalloc.h    | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/js/public/Utility.h b/js/public/Utility.h
index 75214c32..f50fd8dd 100644
--- a/js/public/Utility.h
+++ b/js/public/Utility.h
@@ -77,7 +77,7 @@ enum ThreadType {
 # if defined(DEBUG) || defined(JS_OOM_BREAKPOINT)
 extern bool InitThreadType(void);
 extern void SetThreadType(ThreadType);
-extern uint32_t GetThreadType(void);
+extern JS_FRIEND_API(uint32_t) GetThreadType(void);
 # else
 inline bool InitThreadType(void) { return true; }
 inline void SetThreadType(ThreadType t) {};
diff --git a/js/src/jsalloc.h b/js/src/jsalloc.h
index b9ae5190..234ea9dc 100644
--- a/js/src/jsalloc.h
+++ b/js/src/jsalloc.h
@@ -17,6 +17,8 @@
 #include "js/TypeDecls.h"
 #include "js/Utility.h"
 
+extern JS_PUBLIC_API(void) JS_ReportOutOfMemory(JSContext* cx);
+
 namespace js {
 
 enum class AllocFunction {
@@ -130,7 +132,7 @@ class TempAllocPolicy
 
     bool checkSimulatedOOM() const {
         if (js::oom::ShouldFailWithOOM()) {
-            js::ReportOutOfMemory(reinterpret_cast<ExclusiveContext*>(cx_));
+            JS_ReportOutOfMemory(reinterpret_cast<JSContext*>(cx_));
             return false;
         }
 
-- 
2.11.0 (Apple Git-81)

